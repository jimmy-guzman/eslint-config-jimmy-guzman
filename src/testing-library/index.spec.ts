import testingLibrary from './index'

describe('testing-library', () => {
  it('should export react rules config', () => {
    expect(Object.keys(testingLibrary.overrides[0].rules)).toHaveLength(37)
    expect(testingLibrary).toMatchSnapshot()
  })
})
