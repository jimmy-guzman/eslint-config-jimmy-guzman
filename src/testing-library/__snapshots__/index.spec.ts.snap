// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing-library should export react rules config 1`] = `
Object {
  "overrides": Array [
    Object {
      "files": Array [
        "**/__tests__/**/*.[jt]s?(x)",
        "**/?(*.)+(spec|test).[jt]s?(x)",
      ],
      "plugins": Array [
        "testing-library",
        "jest-dom",
      ],
      "rules": Object {
        "jest-dom/prefer-checked": "error",
        "jest-dom/prefer-empty": "error",
        "jest-dom/prefer-enabled-disabled": "error",
        "jest-dom/prefer-focus": "error",
        "jest-dom/prefer-in-document": "error",
        "jest-dom/prefer-required": "error",
        "jest-dom/prefer-to-have-attribute": "error",
        "jest-dom/prefer-to-have-class": "error",
        "jest-dom/prefer-to-have-style": "error",
        "jest-dom/prefer-to-have-text-content": "error",
        "jest-dom/prefer-to-have-value": "error",
        "testing-library/await-async-query": "error",
        "testing-library/await-async-utils": "error",
        "testing-library/await-fire-event": "error",
        "testing-library/consistent-data-testid": "off",
        "testing-library/no-await-sync-events": "error",
        "testing-library/no-await-sync-query": "error",
        "testing-library/no-container": "error",
        "testing-library/no-debugging-utils": "error",
        "testing-library/no-dom-import": "error",
        "testing-library/no-manual-cleanup": "error",
        "testing-library/no-node-access": "error",
        "testing-library/no-promise-in-fire-event": "error",
        "testing-library/no-render-in-setup": "error",
        "testing-library/no-unnecessary-act": "error",
        "testing-library/no-wait-for-empty-callback": "error",
        "testing-library/no-wait-for-multiple-assertions": "error",
        "testing-library/no-wait-for-side-effects": "error",
        "testing-library/no-wait-for-snapshot": "error",
        "testing-library/prefer-explicit-assert": "off",
        "testing-library/prefer-find-by": "error",
        "testing-library/prefer-presence-queries": Array [
          "error",
          Object {
            "absence": true,
            "presence": true,
          },
        ],
        "testing-library/prefer-query-by-disappearance": "error",
        "testing-library/prefer-screen-queries": "error",
        "testing-library/prefer-user-event": "error",
        "testing-library/prefer-wait-for": "error",
        "testing-library/render-result-naming-convention": "off",
      },
    },
  ],
}
`;
